### ISSUE TITLE
Source directories starting by `_` are ignored

### ISSUE THREAD

**Issue by samoht** [janestreet/jbuilder#185]
Given the following project:
```
$ tree
.
├── _a
│   ├── a.ml
│   └── jbuild
└── a
    ├── a.ml
    └── jbuild

2 directories, 4 files

$ cat a/jbuild
(executable ((name a)))

$ cat _a/jbuild
(executable ((name a)))
```
and `a.ml` are empty files.

The following (unexpected result):

```
$ jbuilder build a/a.exe
    ocamldep a/a.depends.ocamldep-output
      ocamlc a/a.{cmi,cmo,cmt}
    ocamlopt a/a.{cmx,o}
    ocamlopt a/a.exe
✔ 16:13 /tmp/kk $ jbuilder build _a/a.exe
Don't know how to build _a/a.exe
```

**Comment by rgrinberg** [comment]
I believe that's actually intended. Originally it came out my request to ignore `_opam` directories. I think diml then extended it to just ignore any dir prefixed with `_`. It does seem a bit heavy handed though.

**Comment by diml** [comment]
The reason for ignoring directories starting with an underscore is that they often seem to be used for data that we shouldn't look at (_build, _opam, _darcs, ...). @samoht why do you have such a directory?

**Comment by samoht** [comment]
It was a corner case in a test case for the mirage tool (where I create a temporary build directory to build an application). I think it's ok to keep it like this, it was just a very surprising.

**Comment by bmillwood** [comment]
Maybe the right thing to do is print a warning when directories with underscores are in the targets list, or something like that?

**Comment by samoht** [comment]
Yea, that should be enough.

**Comment by diml** [comment]
Seems good to me

**Comment by rgrinberg** [comment]
I think I have a better solution for this. Using #1517, we can make `:standard` mean the current predicate. And you could unignore directories using:

```
(ignored_subdirs (:standard \ _blarg))
```

It's just a bit of a shame that we're specifying the list of things to exclude, rather than a list of things to include.

@diml @samoht is this satisfactory?

**Comment by diml** [comment]
yep, LGTM. I feel like soon we are going to need something as powerful as .gitignore, i.e. to select what should be ignored in sub-directories as well.

**Comment by rgrinberg** [comment]
>  i.e. to select what should be ignored in sub-directories as well.

Yeah, this would be quite useful for ignoring things in the workspace file.

**Comment by anmonteiro** [comment]
Did this stop being possible now that `ignore_dubdirs` is deprecated? I haven't been able to include a directory called `src/__tests__` for dune

**Comment by jeremiedimino** [comment]
> EDIT: (dirs :standard __tests__) seems to work, but only if it's an immediate subdirectory.

Seems related to the discussion in #4216, i.e. write something that applies in every single directory.
